lib/hooks/core/hook-http.js

```diff
@@ -43,19 +43,19 @@ function requestWrap(request) {
 
     var root = cls.getRootContext();
 
+    options = (typeof options === 'string') ?
+      url.parse(options) : clone(options);
+    options.headers = options.headers || {};
+    var uri = uriFromOptions(options);
```

Comment:
We are increasing the path length for the case where `root` is the `nullSpan` (common case) for the benefit of logging. With some refactoring you should be able to avoid this. You might want to outline this into a separate function that you can call from within the `if (!root) {` test.